File: Source/Classic (2010)/Odyssey/UX Kit/Visual Basic/AlphaMessageBoxHandler.vb
Imports System.Windows

Public Class AlphaMessageBoxHandler

    Private m_MessageBoxStyle As MessageBoxButton
    Public Property MessageBoxStyle() As MessageBoxButton
        Get
            Return m_MessageBoxStyle
        End Get
        Set(ByVal value As MessageBoxButton)
            m_MessageBoxStyle = value
        End Set
    End Property

    Private m_MessageBoxResult As MessageBoxResult
    Public Property MessageResult() As MessageBoxResult
        Get
            Return m_MessageBoxResult
        End Get
        Set(ByVal value As MessageBoxResult)
            m_MessageBoxResult = value
        End Set
    End Property

    Function Show(ByVal MessageText As String, ByVal MessageTitle As String, ByVal DialogStyle As MessageBoxButton) As MessageBoxResult
        Dim f As New AlphaMessageBox
        f.alphaMessageBoxStyle = DialogStyle
        f.Title = MessageTitle
        f.textPrompt.Text = MessageText
        f.ShowDialog()
        Dim temp As MessageBoxResult = f.MessageResult
        f.Close()
        Return temp
    End Function

End Class

File: Source/Classic (2010)/Odyssey/Usage Sample/C Sharp/AlphaMessageBoxHandler.cs
using System;
using System.Windows;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Odyssey
{
    public class AlphaMessageBoxHandler
    {

        public MessageBoxButton MessageBoxStyle { get; set; }
        public MessageBoxResult MessageResult { get; set; }

        public MessageBoxResult Show(string MessageText, string MessageTitle, MessageBoxButton DialogStyle) 
        {
            AlphaMessageBox f = new AlphaMessageBox();
            
            f.alphaMessageBoxStyle = DialogStyle;
            f.Title = MessageTitle;
            f.textPrompt.Text = MessageText;
            f.ShowDialog();

            MessageBoxResult temp = f.MessageResult;
            f.Close();
            return  MessageBoxResult.Cancel;
        }
    }
}
